dohq_artifactory/auth.py

```diff
@@ -4,15 +4,26 @@
 class XJFrogArtApiAuth(AuthBase):
     """Attaches X-JFrog-Art-Api Authentication to the given Request object."""
 
-    def __init__(self, apikey):
+    def __init__(self, apikey=None, token=None):
+        if apikey and token:
+            raise RuntimeError("Only one authentication method but get 2")
         self.apikey = apikey
+        self.token = token
 
     def __eq__(self, other):
-        return all([self.apikey == getattr(other, "apikey", None)])
+        return all(
+            [
+                self.apikey == getattr(other, "apikey", None),
+                self.token == getattr(other, "token", None),
+            ]
+        )
 
     def __ne__(self, other):
         return not self == other
 
     def __call__(self, r):
-        r.headers["X-JFrog-Art-Api"] = self.apikey
+        if self.apikey:
+            r.headers["X-JFrog-Art-Api"] = self.apikey
+        else:
```

Comment:
Shouldn't we add one more `if self.token`? 
Right now it looks like we always will add `Authorization` header, even if we don't define it. Just to make it more clear and readable

Reply:
Now, we have a new class for token, and this issue is gone as well.